CryptomatorCommon/Sources/CryptomatorCommonCore/Manager/MasterkeyCacheManager.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MasterkeyCacheManager.swift
+//  CryptomatorCommonCore
+//
+//  Created by Philipp Schmid on 11.01.22.
+//  Copyright © 2022 Skymatic GmbH. All rights reserved.
+//
+
+import CommonCrypto
+import CryptomatorCryptoLib
+import Foundation
+
+public protocol MasterkeyCacheManager {
+	func cacheMasterkey(_ masterkey: Masterkey, forVaultUID vaultUID: String) throws
+	func getMasterkey(forVaultUID vaultUID: String) throws -> Masterkey?
+	func removeCachedMasterkey(forVaultUID vaultUID: String) throws
+}
+
+protocol MasterkeyCacheHelper {
+	func shouldCacheMasterkey(forVaultUID vaultUID: String) -> Bool
+}
+
+public class MasterkeyCacheKeychainManager: MasterkeyCacheManager {
+	public static let shared = MasterkeyCacheKeychainManager(keychain: CryptomatorKeychain(service: "masterkeyCache"))
+	private let keychain: CryptomatorKeychainType
+
+	init(keychain: CryptomatorKeychainType) {
+		self.keychain = keychain
+	}
+
+	public func cacheMasterkey(_ masterkey: Masterkey, forVaultUID vaultUID: String) throws {
+		let aesMasterkey = Array(masterkey.rawKey[0 ..< kCCKeySizeAES256])
+		let macMasterKey = Array(masterkey.rawKey[kCCKeySizeAES256 ..< (2 * kCCKeySizeAES256)])
```

Comment:
As I've already mentioned in our internal review, I wonder if we should make changes in cryptolib-swift so that we don't have to work with ranges (and don't have to import `CommonCrypto`).

We could either make `aesMasterkey` and `macMasterKey` accessible in [Masterkey](https://github.com/cryptomator/cryptolib-swift/blob/986fd4abf8003c176ae050c28a83038222e4ebec/Sources/CryptomatorCryptoLib/Masterkey.swift#L13-L17). I mean, the `rawKey` already has to be public so that it can be used for signing/verifying the vault config.

Or we could add a new static initializer (or change the current one) [`createFromRaw()`](https://github.com/cryptomator/cryptolib-swift/blob/986fd4abf8003c176ae050c28a83038222e4ebec/Sources/CryptomatorCryptoLib/Masterkey.swift#L47-L58), which takes only one argument instead of two. Btw, this initializer is only used for tests. It is public because cloud-access-swift and this project use it for their tests as well.

Any thoughts on that, @overheadhunter?

Reply:
If the [createFromRaw()](https://github.com/cryptomator/cryptolib-swift/blob/986fd4abf8003c176ae050c28a83038222e4ebec/Sources/CryptomatorCryptoLib/Masterkey.swift#L47-L58) initializer is used only for tests we could restrict the visibility at least to internal instead of public as its still possible to access an internal symbol if we import `CryptomatorCryptoLib` with `@testable` (`@testable import CryptomatorCryptoLib`).